easypy/threadtree.py

```diff
@@ -20,21 +19,29 @@
 
 
 _REGISTER_GREENLETS = False
+_FRAME_SNAPSHOTS_REGISTRY = defaultdict(dict)
+
 _orig_start_new_thread = _thread.start_new_thread
 
 
 def start_new_thread(target, *args, **kwargs):
     """
-    A wrapper for the built in 'start_new_thread' used to capture the parent of each new thread.
+    A wrapper for the built-in 'start_new_thread' used to capture the parent of each new thread.
     """
-    parent_uuid = get_thread_uuid()
+    parent_thread = threading.current_thread()
+    parent_uuid = get_thread_uuid(parent_thread)
+    parent_frame = sys._getframe(0)
```

Comment:
The performance impact of creating frame snapshot is neglectable

<img width="1057" alt="Create snapshots" src="https://github.com/user-attachments/assets/6390c43e-95ee-4e21-affd-bd683e1b386d" />